src/main/java/com/github/jparkie/pdd/BitArray.java

```diff
@@ -38,22 +38,35 @@ private static int numWords(long numBits) {
         return (int) numWords;
     }
 
+    private final class indexAndMask {
+        public final int index;
+        public final long mask;
+
+        public indexAndMask(long index) {
+            this.index = (int)(index >>> 6);
+            this.mask = 1L << (index & 0x3F);
+        }
+    }
+
     public boolean get(long index) {
-        return (data[(int) (index >>> 6)] & (1L << index)) != 0;
+        final indexAndMask iam = new indexAndMask(index);
+        return (data[iam.index] & iam.mask) != 0;
     }
 
     public boolean set(long index) {
-        if (!get(index)) {
```

Comment:
My thought of calling `get()` in `set()` and `clear()` was that the JVM would inline the `get()` call and optimize the arithmetic operations.